### PR TITLE
[cgroups2] Add cgroups2::path to get the path of a cgroup.

### PR DESCRIPTION
**Pull request by DevinLeamy**
Introduces a utility to get the absolute path of a cgroup and refactors code to use it.

### CODE REVIEW

File: src/tests/containerizer/cgroups2_tests.cpp [hunk 1]
```diff
@@ -115,6 +115,14 @@ TEST_F(Cgroups2Test, ROOT_CGROUPS2_AssignProcessToCgroup)
   AWAIT_EXPECT_WTERMSIG_EQ(SIGKILL, process::reap(pid));
 }
 
+
+TEST_F(Cgroups2Test, CGROUPS2_Path)
+{
+  EXPECT_EQ("/sys/fs/cgroup/", cgroups2::path(cgroups2::ROOT_CGROUP));
+  EXPECT_EQ("/sys/fs/cgroup/foo", cgroups2::path("foo"));
+  EXPECT_EQ("/sys/fs/cgroup/foo/bar", cgroups2::path("foo/bar"));
```
Comment (DevinLeamy) [inline src/tests/containerizer/cgroups2_tests.cpp:123]: The matches the behavior of python3s `os.path.join`.

File: src/linux/cgroups2.hpp [hunk 2]
```diff
@@ -75,6 +75,10 @@ Try<Nothing> move_process(const std::string& cgroup, pid_t pid);
 // /sys/fs/cgroup. E.g. For /sys/fs/cgroup/test, this will return "test".
 Try<std::string> cgroup(pid_t pid);
 
+
+// Get the absolute of a cgroup. Assumes to cgroup exists.
```
Comment (bmahler) [inline src/linux/cgroups2.hpp:79]: s/to/the/

what does this assumption entail though?

File: src/tests/containerizer/cgroups2_tests.cpp [hunk 3]
```diff
@@ -115,6 +115,14 @@ TEST_F(Cgroups2Test, ROOT_CGROUPS2_AssignProcessToCgroup)
   AWAIT_EXPECT_WTERMSIG_EQ(SIGKILL, process::reap(pid));
 }
 
+
+TEST_F(Cgroups2Test, CGROUPS2_Path)
+{
+  EXPECT_EQ("/sys/fs/cgroup/", cgroups2::path(cgroups2::ROOT_CGROUP));
```
Comment (bmahler) [inline src/tests/containerizer/cgroups2_tests.cpp:121]: I guess this is indicating that you have two ways of calling this function?

1. Pass a "cgroup" relative to the hierarchy, you get back the absolute path.
2. Pass a "cgroup" using an absolute path, the absolute path must include the hierarchy already in this case, e.g. if you pass "/foo", you'll only get back "/foo", not "/sys/fs/cgroup/foo".

File: src/linux/cgroups2.cpp [hunk 4]
```diff
@@ -309,10 +307,10 @@ Try<Nothing> destroy(const string& cgroup)
 
 Try<Nothing> move_process(const string& cgroup, pid_t pid)
 {
-  const string absolutePath = path::join(MOUNT_POINT, cgroup);
+  const string& path = cgroups2::path(cgroup);
```
Comment (bmahler) [inline src/linux/cgroups2.cpp:310]: we can remove the ref here to be consistent with the other calls

**Pull request closed without merging**